pkg/client/object.go

```diff
@@ -0,0 +1,881 @@
+		return c.getObjectV2(ctx, p, opts...)
+	default:
+		return nil, unsupportedProtocolErr
+	}
+}
+
+func (c *Client) getObjectV2(ctx context.Context, p *GetObjectParams, opts ...CallOption) (*object.Object, error) {
+	// create V2 Object client
+	cli, err := v2ObjectClient(c.remoteNode.Protocol, c.opts)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not create Object V2 client")
+	}
+
+	callOpts := defaultCallOptions()
+
+	for i := range opts {
+		if opts[i] != nil {
+			opts[i].apply(&callOpts)
+		}
+	}
+
+	// create request
+	req := new(v2object.GetRequest)
+
+	// initialize request body
+	body := new(v2object.GetRequestBody)
+	req.SetBody(body)
+
+	// set meta header
+	req.SetMetaHeader(v2MetaHeaderFromOpts(callOpts))
+
+	// fill body fields
+	body.SetAddress(p.addr.ToV2())
+
+	// sign the request
+	if err := signature.SignServiceMessage(c.key, req); err != nil {
+		return nil, errors.Wrapf(err, "could not sign %T", req)
+	}
+
+	// create Get object stream
+	stream, err := cli.Get(ctx, req)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not create Get object stream")
+	}
+
+	var (
+		payload []byte
+		obj     = new(v2object.Object)
+	)
+
+	for {
+		// receive message from server stream
+		resp, err := stream.Recv()
+		if err != nil {
+			if errors.Is(errors.Cause(err), io.EOF) {
+				break
+			}
+
+			return nil, errors.Wrap(err, "could not receive Get response")
+		}
+
+		// verify response structure
+		if err := signature.VerifyServiceMessage(resp); err != nil {
+			return nil, errors.Wrapf(err, "could not verify %T", resp)
+		}
+
+		switch v := resp.GetBody().GetObjectPart().(type) {
+		case nil:
+			return nil, errors.New("received nil object part")
+		case *v2object.GetObjectPartInit:
+			obj.SetObjectID(v.GetObjectID())
+			obj.SetSignature(v.GetSignature())
+
+			hdr := v.GetHeader()
+			obj.SetHeader(hdr)
+
+			if p.w == nil {
+				payload = make([]byte, 0, hdr.GetPayloadLength())
+			}
+		case *v2object.GetObjectPartChunk:
+			if p.w != nil {
+				if _, err := p.w.Write(v.GetChunk()); err != nil {
+					return nil, errors.Wrap(err, "could not write payload chunk")
+				}
+			} else {
+				payload = append(payload, v.GetChunk()...)
+			}
+		default:
+			panic(fmt.Sprintf("unexpected Get object part type %T", v))
+		}
+	}
+
+	obj.SetPayload(payload)
+
+	// convert the object
+	res, err := object.FromV2(obj)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not convert V2 object")
+	}
+
+	return res, nil
+}
+
+func (p *ObjectHeaderParams) WithAddress(v *object.Address) *ObjectHeaderParams {
+	if p != nil {
+		p.addr = v
+	}
+
+	return p
+}
+
+func (p *ObjectHeaderParams) WithAllFields() *ObjectHeaderParams {
+	if p != nil {
+		p.short = false
+	}
+
+	return p
+}
+
+func (p *ObjectHeaderParams) WithMainFields() *ObjectHeaderParams {
+	if p != nil {
+		p.short = true
+	}
+
+	return p
+}
+
+func (c *Client) GetObjectHeader(ctx context.Context, p *ObjectHeaderParams, opts ...CallOption) (*object.Object, error) {
```

Comment:
Option for short / full header will be set at `*ObjectHeaderPararms`?

Reply:
Yes, through `WithAllFields`, `WithMainFields` methods.